src/utils/cip14.rs

```diff
@@ -0,0 +1,120 @@
+//! An utility to build fingerprints for Cardano assets CIP14
+//! https://cips.cardano.org/cips/cip14/
+//! 
+use cryptoxide::{digest::Digest, blake2b::Blake2b};
+use crate::utils::bech32::{Bech32Config,Bech32Provider};
+
+pub fn blake2b160(data: &[u8]) -> [u8;20] {
```

Comment:
The code already exists in https://github.com/txpipe/pallas/blob/main/pallas-crypto/src/hash/hasher.rs.

You just have to add:
```
common_hasher!(160);
```
at https://github.com/txpipe/pallas/blob/1482303616e9ba68f322eceb9ed03118efee28d5/pallas-crypto/src/hash/hasher.rs#L109 and some comments in the header, then you can do something like:

```
use pallas::crypto::hash::{Hash, Hasher},

fn asset_fingerprint(policy: &ByteVec, asset: &ByteVec) -> String {
    let mut hasher = Hasher::<160>::new();
    hasher.input(policy);
    hasher.input(asset);
    bech32::encode(
        &"asset".to_string(),
        hasher.finalize().to_base32(),
        bech32::Variant::Bech32,
    )
    .unwrap()
}
```
Likely returning a Result instead of the `unwrap` here.

Reply:
I also think we might want the fingerprint in the `MintRecord` struct too, but this kind of leads to duplicates, so I'm not sure what's the best option here.